system/Helpers/array_helper.php

```diff
@@ -80,3 +80,195 @@ function _array_search_dot(array $indexes, array $array)
 		return $array[$currentIndex];
 	}
 }
+
+if (!function_exists('sort_array_by_column'))
+{
+	/**
+	 * Sort an array by column name in multidimensional array
+	 *
+	 * @param array  $array
+	 * @param string $column
+	 * @param int    $direction
+	 * @param int    $sort_type
+	 *
+	 * @return void
+	 */
+	function sort_array_by_column(array &$array, string $column, int $direction = SORT_ASC, int $sort_type = SORT_REGULAR)
+	{
+		$sort_col = [];
+		foreach ($array as $key => $row)
+		{
+			$sort_col[$key] = $row[$column];
+		}
+		array_multisort($sort_col, $direction, $array, $sort_type);
+	}
+}
+
+if (!function_exists('reduce_array'))
+{
+	/**
+	 * Eliminate the keys from single dimensional array
+	 * 
+	 * @param array $array
+	 * @param array $keys
+	 *
+	 * @return array
+	 */
+	function reduce_array(array $array, array $keys)
+	{
+		$return_array = [];
+		foreach ($array as $key => $value)
+		{
+			if (!in_array($key, $keys))
+			{
+				$return_array[$key] = $value;
+			}
+		}
+
+		return $return_array;
+	}
+}
+
+if (!function_exists('array_non_empty_items'))
+{
+	/**
+	 * Return all non-empty elements of array
+	 * 
+	 * @param array|string $array
+	 *
+	 * @return array|string
+	 */
+	function array_non_empty_items($array)
```

Comment:
I'm confused how this does anything different than array_filter on it's own can do?